kernels/simple_copy/main.c

```diff
@@ -0,0 +1,60 @@
+#include "data.h"
+#include "memref.h"
+#include "stdint.h"
+#include <snrt.h>
+
+// treat incoming size as pointer
+void _mlir_ciface_snax_dma_1d_transfer(size_t *source, size_t *destination,
+                                       size_t *size) {
+  snrt_dma_start_1d((void *)destination, (void *)source,
+                    *size * sizeof(size_t));
+  return;
+}
```

Comment:
Let's move this to a runtime library at some point.
Okay to keep it here now for LLD fun